controllers/azuremachine_controller.go

```diff
@@ -246,10 +245,6 @@ func (r *AzureMachineReconciler) reconcileNormal(ctx context.Context, machineSco
 		machineScope.SetFailureMessage(errors.Errorf("Azure VM state %q is unexpected", vm.State))
 	}
 
-	if err := ams.reconcileNetworkInterface(azure.GenerateNICName(machineScope.Name())); err != nil {
```

Comment:
this was being called twice (once here and once in azuremachine_reconciler before VM create)